### PR TITLE
[visionOS] Fix visionOS build errors on Xcode 15.0 Beta 2 

### PR DESCRIPTION
**Pull request by ncooke3**
Tracking open questions in https://github.com/google/GoogleUtilities/issues/109

### CODE REVIEW

File: GoogleUtilities/Tests/Unit/Swizzler/GULAppDelegateSwizzlerTest.m [hunk 1]
```diff
@@ -871,6 +871,9 @@ - (void)testApplicationDidFailToRegisterForRemoteNotificationsIsInvokedOnInterce
   XCTAssertEqual(testAppDelegate.failToRegisterForRemoteNotificationsError, error);
 }
 
+// TODO: Re-enable and adjust accordingly after answering API availability
+// questions outlined in issue #109.
+#if TARGET_OS_IOS && (!defined(TARGET_OS_XR) || !TARGET_OS_XR)
```
Comment (paulb777) [inline GoogleUtilities/Tests/Unit/Swizzler/GULAppDelegateSwizzlerTest.m:876]: I believe this is testing the notification APIs that are deprecated on iOS as well after iOS 10, so I'm not sure there is any need to re-enable.

Comment (andrewheard) [reply GoogleUtilities/Tests/Unit/Swizzler/GULAppDelegateSwizzlerTest.m:876]: I think you're right, I see this is explicitly marked as unavailable on visionOS:
```
- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo
  API_DEPRECATED("Use UserNotifications Framework's -[UNUserNotificationCenterDelegate willPresentNotification:withCompletionHandler:] or -[UNUserNotificationCenterDelegate didReceiveNotificationResponse:withCompletionHandler:] for user visible notifications and -[UIApplicationDelegate application:didReceiveRemoteNotification:fetchCompletionHandler:] for silent remote notifications",
  ios(3.0, 10.0)) API_UNAVAILABLE(xros);
```

Comment (ncooke3) [reply GoogleUtilities/Tests/Unit/Swizzler/GULAppDelegateSwizzlerTest.m:876]: Thank y'all, I'll adjust.

**Comment by ncooke3** [comment]
This should be gtg on green CI.

**Review by andrewheard — APPROVED** [review]

**Pull request merged**